third_party/nvfuser/csrc/fusion.cpp

```diff
@@ -226,6 +226,12 @@ void Fusion::addOutput(Val* output) {
   // unfortunately would call addOutput after we marked io_alias_ map.
   // TORCH_CHECK(io_alias_.count(output) == 0,
   //     "can't register aliased output as real output");
+  if (output->isFusionInput() && io_alias_.find(output) == io_alias_.end()) {
+    // If explicitly adding an input as an output, replace with call to
+    // aliasOutputToInput
+    aliasOutputToInput(output, output);
+    return;
+  }
```

Comment:
I'm afraid I have no idea what the comment above means or if it's still relevant. 
@jjsjann123, does this change make sense?

Reply:
hmm... good question!

Did this actually work?! Since we are adding a short-cut for aliased output, looks like we are skipping
`  outputs_.push_back(output);`. Doesn't this remove the kernel arguments?

I'm not sure if we actually care about setting memory type of tv any more, since we don't need to allocate anything.

Reply:
We actually don't skip that part, since `aliasOutputToInput` will call `addOutput` again, and this time find `output` in `io_alias_`, so this if block will not fire.

Reply:
Oops. sorry I mis-read this one. Looking good to me.